### PR TITLE
Fix clang error and warnings

### PR DESCRIPTION
**Pull request by Dr15Jones**
This fixes a clang link time error and two warnings.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
@davidlt thought you'd want to know about this

**Comment by davidlt** [comment]
This should obsolete my PR then. This one is better.

**Comment by slava77** [comment]
#19239 should be closed then

**Comment by Dr15Jones** [comment]
@davidlt sorry, I hadn't seen yours.

File: RecoParticleFlow/PFClusterProducer/plugins/SimMappers/GenericSimClusterMapper.cc [hunk 1]
```diff
@@ -43,7 +43,7 @@ buildClusters(const edm::Handle<reco::PFRecHitCollection>& input,
     reco::PFCluster& back = output.back();
     edm::Ref<std::vector<reco::PFRecHit> > seed;    
     double energy = 0.0, highest_energy = 0.0;
-    auto hitsAndFractions = std::move( sc.hits_and_fractions() );
+    auto hitsAndFractions = sc.hits_and_fractions() ;
```
Comment (slava77) [inline RecoParticleFlow/PFClusterProducer/plugins/SimMappers/GenericSimClusterMapper.cc]: do we have all compilers used now conforming to the copy elision requirement?

Comment (slava77) [inline RecoParticleFlow/PFClusterProducer/plugins/SimMappers/GenericSimClusterMapper.cc]: to ask more correctly: are all of the compilers that we use now in the default optimization applying copy elision?

Comment (Dr15Jones) [inline RecoParticleFlow/PFClusterProducer/plugins/SimMappers/GenericSimClusterMapper.cc]: In both cases where I removed `std::move` the clang compiler warning was:
```
warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
     auto hitsAndFractions = std::move( sc.hits_and_fractions() );
```

i.e. in both cases it was a temporary anyway so `std::move` was not useful even if copy elision is not available.

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/19248	32375fb
- reco update is to allow a clear copy elision (in RecoParticleFlow/PFClusterProducer); a technical update
- jenkins tests pass (Reco comparison results: 0 differences found in the comparisons)

**Comment by davidlange6** [comment]
merge

**Pull request merged**